pkg/transfers/inbound/cleanup.go

```diff
@@ -54,3 +78,43 @@ func deleteFilesOnRemote(logger log.Logger, agent upload.Agent, localDir, suffix
 	}
 	return el
 }
+
+// deleteEmptyFiles deletes all empty files that are older than after (time.Duration)
+func deleteEmptyFiles(logger log.Logger, agent upload.Agent, localDir, suffix string, now time.Time, after time.Duration) error {
+	baseDir := filepath.Join(localDir, suffix)
+	infos, err := ioutil.ReadDir(baseDir)
+	if err != nil {
+		return fmt.Errorf("reading %s: %v", baseDir, err)
+	}
+
+	var el base.ErrorList
+	for i := range infos {
+		fileInfo := infos[i]
+		path := filepath.Join(suffix, filepath.Base(infos[i].Name()))
+		if shouldDeleteEmptyFile(fileInfo, now, after) {
+			err := agent.Delete(path)
+			if err != nil {
+				el.Add(err)
+			} else {
+				logger.Log(fmt.Sprintf("deleted zero byte file %s", path))
+			}
+		} else {
+			logger.Log(fmt.Sprintf("zero byte file not deleted %s", path))
+		}
```

Comment:
I recommend doing something like the following to reduce the nested if-else blocks
```go		
if !shouldDeleteEmptyFile(fileInfo, now, after) {
  continue
}
if err := agent.Delete(path); err != nil {
		el.Add(err)
}
logger.Logf("deleted zero byte file %s", path)
}
```

Reply:
The nested if-elses totally suck. Let me see if I can clean it up with something like you suggested.

Reply:
heh yeah, with Go errors-handling, a good rule of thumb is to always "exit" early

Reply:
Pushed.